engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/SqlSourceFactory.scala

```diff
@@ -0,0 +1,70 @@
+package pl.touk.nussknacker.engine.flink.table.source
+
+import pl.touk.nussknacker.engine.api.component.ProcessingMode
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.context.transformation.{
+  DefinedEagerParameter,
+  NodeDependencyValue,
+  SingleInputDynamicComponent
+}
+import pl.touk.nussknacker.engine.api.definition._
+import pl.touk.nussknacker.engine.api.process.{BasicContextInitializer, Source, SourceFactory}
+import pl.touk.nussknacker.engine.api.{NodeId, Params}
+import pl.touk.nussknacker.engine.flink.table.SqlDataSourcesDefinition
+import pl.touk.nussknacker.engine.flink.table.extractor.DataSourceTableDefinition
+
+class SqlSourceFactory(defs: SqlDataSourcesDefinition) extends SingleInputDynamicComponent[Source] with SourceFactory {
+
+  override type State = DataSourceTableDefinition
+
+  private val tableNameParamName = "Table"
+
+  private val tableNameParam: ParameterWithExtractor[String] = {
+    val possibleTableParamValues =
+      defs.tableDefinitions.map(c => FixedExpressionValue(s"'${c.tableName}'", c.tableName))
```

Comment:
We usually adds an extra "non-picked" option which is the default option. Thanks to that we avoid situation that we pick the table that user don't want to pick. Take a look at other places of `FixedValuesParameterEditor` usage.